terraform-beyond-layer0/example-3/env-dev/variables.tf

```diff
@@ -0,0 +1,30 @@
+#
+# example-3/env-dev/variables.tf
+#
+
+variable "endpoint" {}
+
+variable "token" {}
+
+variable "access_key" {}
+
+variable "secret_key" {}
+
+variable "environment" {
+  default = "dev"
+}
+
+variable "region" {
+  default = "us-west-2"
+}
+
+# rather than add conditional statements to resources configurations
+# creating a lookup table can be more useful
+variable "service_scale" {
```

Comment:
It is used in main.tf and provides a default that can be modified by the consumer
```
module "guestbook" {
  source = "./../modules/guestbook_service"
  ...
  service_scale = "${var.service_scale}"
}
```